File: python/math/algorithms/algo1/bitWise/bitWise.py

def main():
    maxAND = 0
    maxOR = 0
    maxXOR = 0

    n = int(input("Enter a number: "))
    k = int(input("Enter a number: "))

    for i  in range(1,k+1):
        for j in range(i+1,n+1):
            nAndK = i & j
            nOrK = i | j
            nXorK = i ^ j
            print("i=", i , "; j=", j, "; n&k=", nAndK, "; n|k= ", nOrK, "; n xor k=",nXorK)
            if((nAndK > maxAND) and (nAndK < k)):
                maxAND = nAndK
            if((nOrK > maxOR) and (nOrK < k)):
                maxOR = nOrK
            if((nXorK > maxXOR) and (nXorK < k)):
                maxXOR = nXorK
    
    print("maxAND= ", maxAND, "; maxOR= ", maxOR, "; maxXOR- ", maxXOR)

if __name__ == "__main__":
    main()

File: java/eclipse/hackerRank/bitWise/bitWise/src/bitWise/BitWise.java
package bitWise;

// Needed for the Scanner class for input
import java.util.Scanner;

public class BitWise {

	public static void main(String[] args) {
		
		int n;
		int k;
		
		Scanner input = new Scanner (System.in);
		// Ask user input the first integer
		System.out.println("Enter the first integer: ");
		n = input.nextInt();
		
		// Ask user input the second integer
		System.out.println("Enter the second integer: ");
		k = input.nextInt();
		
		calculateMaximumValues(n,k);
		
		input.close();

	}
	
	public static void calculateMaximumValues(int n, int k) {
		int i,j;
		int maxAND = 0;
		int maxOR = 0;
		int maxXOR = 0;
		int nAndK = 0;
		int nOrK = 0;
		int nXorK = 0;
		
		for (i=1; i<=k; i++)
		{
			for (j=i+1; j<=n; j++)
			{
				nAndK = i&j;
				nOrK = i|j;
				nXorK = i^j;
				
				System.out.printf("n=%d; k=%d, n&k=%d; n|k=%d; n xor k=%d\n", i,j,nAndK,nOrK,nXorK);
				
				if((nAndK > maxAND) && (nAndK < k))
					maxAND = nAndK;
				if((nOrK > maxOR) && (nOrK < k))
					maxOR = nOrK;
				if((nXorK > maxXOR) && (nXorK < k))
					maxXOR = nXorK;
			}
		}
		
		System.out.printf("maxAND = %d; maxOR = %d; maxXOR = %d\n", maxAND, maxOR, maxXOR);
		
	}

}
